kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -182,6 +182,10 @@ private[kyuubi] class EngineRef(
 
     conf.set(HA_NAMESPACE, engineSpace)
     conf.set(HA_ENGINE_REF_ID, engineRefId)
+
+    val metadataStoreConf = conf.getAll.filter(x => x._1.startsWith(KYUUBI_METADATA_STORE_PREFIX))
```

Comment:
I suppose we can add those configurations to `KyuubiConf#serverOnlyConfEntries` to solve this problems.

Reply:
The metadata sore related configs are defined in `org.apache.kyuubi.server.metadata.jdbc.JDBCMetadataStoreConf._`, which is in kyuubi-server. KyuubConf is in kyuubi-common and has no access to JDBCMetadataStoreConf. Any solution for this?

Reply:
Emm... how about
1. making `serverOnlyConfEntries` mutable
2. adding method `KyuubiConf#registerServerOnlyConfEntry`
3. invoking `KyuubiConf#registerServerOnlyConfEntry` in `JDBCMetadataStoreConf`

Reply:
what about making `serverOnly` a property for `ConfigEntry` and `ConfigBuilder`? just like `internal`

Reply:
sgtm